### PR TITLE
Added tests engine

### PR DESCRIPTION
**Pull request by nmamro**

### CODE REVIEW

File: GUITests/Tests/SignInTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using Kernel;
+using Kernel.Enums;
+using OpenQA.Selenium.Chrome;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace GUITestsEngine.Tests
+{
+    public class SignInTests
```
Comment (dTarasova) [inline GUITests/Tests/SignInTests.cs:13]: Why do sign in tests use create user request?

File: GUITests/Core/Engine.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            });
+        }
+
+        private static void RunTestMethod(Type type, MethodInfo method)
+        {
+            var ctor = type.GetConstructor(Type.EmptyTypes);
+
+            if (ctor == null)
+            {
+                throw new FormatException($"{type.Name} must have parameterless constructor");
```
Comment (dTarasova) [inline GUITests/Core/Engine.cs:51]: It is new custom exception?

Comment (nmamro) [reply GUITests/Core/Engine.cs:51]: > It is new custom exception?

No

File: GUITests/Tests/SignInTests.cs (hunk 1 above)
Comment (nmamro) [reply GUITests/Tests/SignInTests.cs:13]: > Why do sign in tests use create user request?

To be able to login user with correct data

File: GUITests/Core/Engine.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            var instance = ctor.Invoke(null);
+
+            var isPassed = false;
+
+            try
+            {
+                method.Invoke(instance, null);
+                isPassed = true;
+            }
+            catch { }
```
Comment (denis-kriv) [inline GUITests/Core/Engine.cs:63]: Why do we need this empty "catch block"?

File: GUITests/Core/Engine.cs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            finally
+            {
+                testResults[type].Add(new TestInfo(method.Name, isPassed));
+            }
+        }
+
+        private static void PrintReport()
+        {
+            Console.WriteLine("Testing report:");
+            Console.WriteLine("-----------------------------");
```
Comment (denis-kriv) [inline GUITests/Core/Engine.cs:73]: Why not to create a variable for this string?

**Review by denis-kriv — DISMISSED** [review]

File: GUITests/Core/Engine.cs (hunk 4 above)
Comment (denis-kriv) [reply GUITests/Core/Engine.cs:73]: I meant this "-----------------------------"

File: GUITests/Tests/SignUpTests.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            Thread.Sleep(5000);
+
+            WebDriverWrapper.FindElementByClass(driver, "a", "Sign up").Click();
+
+            WebDriverWrapper.FillForm(driver, input);
+
+            WebDriverWrapper.FindElementByClass(driver, "button", "Next").Click();
+
+            Thread.Sleep(5000);
+        }
```
Comment (Maksim-Trolina) [inline GUITests/Tests/SignUpTests.cs:37]: Bad practice. Better to use explicit expectations.

File: GUITests/Core/Engine.cs (hunk 4 above)
Comment (nmamro) [reply GUITests/Core/Engine.cs:73]: > Why not to create a variable for this string?

Why don't you like this string?

File: GUITests/Core/Engine.cs (hunk 3 above)
Comment (nmamro) [reply GUITests/Core/Engine.cs:63]: > Why do we need this empty "catch block"?

Not to fall after exception is thrown, but IsPassed was created ass false already, so we don't need to reassign its value here

File: GUITests/Core/Engine.cs (hunk 4 above)
Comment (denis-kriv) [reply GUITests/Core/Engine.cs:73]: You use this string too often and you can create a variable, but it is my opinion

**Review by dTarasova — APPROVED** [review]

**Review by dTarasova — APPROVED** [review]

**Review by Cerberous007 — APPROVED** [review]

**Review by Maksim-Trolina — APPROVED** [review]

**Review by ifelseelif — APPROVED** [review]

**Pull request merged**